tags/tags-impl/impl/src/java/org/sakaiproject/tags/impl/TagServiceImpl.java

```diff
@@ -78,15 +88,74 @@ public void destroy() {
     }
 
     @Override
-    public TagCollections getTagCollections() {
-        return tagCollections;
+    public void saveTagAssociation(String itemId, String tagId) {
```

Comment:
I think this save should only do the save... the new and the set should be out of this method

Reply:
I get what you mean, but I did it this way so the different tools that will be integrated don't have to use TagAssociation objects. They just pass the needed params and that's it, and those lines don't need to be repeated.

Tbh the existing tagService is a bit strange/old and I just started using JPA for my new class, but a refactor could be interesting in the near future...

Reply:
Great! No problem